clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/BootstrappingVeniceChangelogConsumer.java

```diff
@@ -41,9 +46,19 @@ public interface BootstrappingVeniceChangelogConsumer<K, V> {
   /**
    * polls for the next batch of change events. The first records returned following calling 'start()' will be from the bootstrap state.
    * Once this state is consumed, subsequent calls to poll will be based off of recent updates to the Venice store.
+   *
+   * In the experimental client, records will be returned in batches configured to the MAX_BUFFER_SIZE. So the initial
+   * calls to poll will be from records from the bootstrap state, until the partitions have caught up.
+   * Additionally, if the buffer hits the MAX_BUFFER_SIZE before the timeout is hit, poll will return immediately.
+   *
    * @param timeoutInMs
    * @return
    */
   Collection<PubSubMessage<K, ChangeEvent<V>, VeniceChangeCoordinate>> poll(long timeoutInMs);
 
+  /**
+   * @return True if all subscribed partitions have caught up.
```

Comment:
I think it might be worth clarifying that once `isCaughtUp` is true it will remain true even if ingestion is slowed/stalled for subscribed partitions and we start to lag.